common/src/main/java/com/wynntils/core/features/overlays/ContainerOverlay.java

```diff
@@ -33,24 +34,39 @@
     private final List<T> children = new ArrayList<>();
     private final Map<T, OverlaySize> inherentSize = new HashMap<>();
 
-    protected ContainerOverlay(OverlayPosition position, OverlaySize size, GrowDirection growDirection, int spacing) {
+    protected ContainerOverlay(
+            OverlayPosition position,
+            OverlaySize size,
+            GrowDirection growDirection,
+            int spacing,
+            HorizontalAlignment horizontalAlignment,
+            VerticalAlignment verticalAlignment) {
         super(position, size);
         this.growDirection.updateConfig(growDirection);
         this.spacing.updateConfig(spacing);
+        this.horizontalAlignmentOverride.updateConfig(horizontalAlignment);
+        this.verticalAlignmentOverride.updateConfig(verticalAlignment);
     }
 
-    protected ContainerOverlay(OverlayPosition position, OverlaySize size, GrowDirection growDirection) {
-        this(position, size, growDirection, DEFAULT_SPACING);
+    protected ContainerOverlay(
+            OverlayPosition position,
+            OverlaySize size,
+            GrowDirection growDirection,
+            HorizontalAlignment horizontalAlignment,
+            VerticalAlignment verticalAlignment) {
+        this(position, size, growDirection, DEFAULT_SPACING, horizontalAlignment, verticalAlignment);
     }
 
     public void addChild(T overlay) {
         inherentSize.put(overlay, overlay.getSize().copy());
         children.add(overlay);
+        WynntilsMod.registerEventListener(overlay);
```

Comment:
You're making all overlays listeners? I guess that is fine, but why? I assumed that I had manually registered the TokenOverlay as a listener and that you would remove that as part of this fix, but I could not find it. Are there any other overlays that are "manually" registered now that does not need to be? If not, why is this needed? 🤔

Reply:
It was you that made all overlays event listeners afaik. But yes, every overlay that works with functions uses events. And some other overlays too, so the holder feature does not have to hook into the overlay. So this is needed.

Reply:
(Registering all overlays caused the unintended issue of https://github.com/Wynntils/Artemis/pull/1446)

Reply:
Ok, I'm apparently pretty confused about this issue. :) Let me rephrase: This seems like you are adding a registration. If you have moved this from somewhere else, there should have been a corresponding delete somewhere in the patch.

Oh. Now I think the penny dropped. This is for Container children; other overlays get registered by the overlay framework. I see.